frontend/src/Editor/QueryManager/Components/QueryManagerHeader.jsx

```diff
@@ -100,7 +100,11 @@ export const QueryManagerHeader = forwardRef(({ darkMode, options, editorRef },
         })}
       >
         <button
-          onClick={() => runQuery(editorRef, selectedQuery?.id, selectedQuery?.name)}
+          onClick={() =>
+            runQuery(editorRef, selectedQuery?.id, selectedQuery?.name, undefined, 'edit', {
+              executeRunQueryFromRunButton: true,
```

Comment:
can you make it something like `shouldSetPreviewData`. Incase if we need to set preview data from somewhere else we can use it. Plus straight forward

Reply:
Make sense, more generalized, and is more readable. Updating it.